pages/reset_password.php

```diff
@@ -3,17 +3,41 @@
 
 if ($_SERVER['REQUEST_METHOD'] == 'POST') {
 	require('../../mysqli_connect.inc.php');
+
+	include('includes/header.html');
+	echo '<div class = "col-sm-6">';
+
 	if (isset($_POST['email'])) {
 		$e = mysqli_real_escape_string($dbc, trim($_POST['email']));
 	} else {
 		$errors[] = "No email";
 	}
 
+	$q = "SELECT email FROM users WHERE email = '$e'";
+	$r = mysqli_query($dbc, $q);
+	if (mysqli_num_rows($r) == 0) {
+		$errors[] = "That is not a registered email. ";
+	}
+
+
 	if (empty($errors)) {
-		include('includes/header.html');
-		echo '<div class = "col-sm-10" id = "reset-password-body">';	
-		echo '<div>An email was sent to your account</div>';	
-		sendEmail('Reset your Password', 'Click the link below to reset your Borum password. ', $e, 'An email was sent to you. Click the link in the email to reset your password.');
+		
+		echo '<div class = "col-sm-10" id = "reset-password-body">';
+		$q = "SELECT SHA2(rand(5000), 512)";
+		$r = mysqli_query($dbc, $q);
+		$row = mysqli_fetch_array($r, MYSQLI_NUM);
+		$q = "INSERT INTO `password-resets` (code, email) VALUES ('{$row[0]}', '$e')";
+		$r = mysqli_query($dbc, $q);
+		sendEmail('Reset your Password', 'Click the link below to reset your Borum password. <br><a href = "http://www.bforborum.com/pages/change_password.php?code=' . $row[0] . '&email=' . $e . '">Click here</a>', $e, 'An email was sent to you. Click the link in the email to reset your password.');
```

Comment:
Excellent! It is concise, seems to work, and looks consistent with all other mysqli queries.